typing/typeclass.ml

```diff
@@ -856,11 +861,12 @@ and class_structure cl_num final val_env met_env loc
   end;
 
   (* Typing of class fields *)
-  let (_, _, _, fields, concr_meths, _, inher, _local_meths, _local_vals) =
+  let class_env = {val_env=val_env; met_env=met_env; par_env=par_env} in
```

Comment:
OCaml has a feature called (record label) "puning" that allows you to write `{val_env; ...}` instead of `{val_env = val_env; ...}` (same for all other labels of course) here. The idea is that, within a record, `x;` is equivalent to `x=x;`, both in record expressions and record patterns.